### DISCUSSION TITLE
Ability change icon of empty state

### DISCUSSION THREAD

**Discussion by Daxble** [icflorescu/mantine-datatable#19] [Ideas]
Ability to change the icon of the empty state would be wonderful. Maybe even the ability to completely replace the empty state entirely.

**Comment by icflorescu** [comment]
Sounds reasonable.

Just thinking aloud here: I think we could add two more optional properties called `noRecordsIcon`and `emptyState`. The latter would take precedence over `noRecordsText` and `noRecordsIcon`.
As a side note, if you provide an `emptyState` or an `emptyIcon`, you'd probably have to make sue they are themed correctly.

What do you think?

**Reply by Daxble** [reply]
Sounds like a good set of options, makes it easier for people who just want to swap out the icon but keep the option open for more advanced use cases. Handling theming ourselves is the best option in my opinion, too many edge cases and don't have to overwrite styles.

**Comment by icflorescu** [comment]
This should be fixed in `v1.4.0`. See [icflorescu.github.io/mantine-datatable/examples/empty-state](https://icflorescu.github.io/mantine-datatable/examples/empty-state).
